### DISCUSSION TITLE
Problem with Youtube RTMP streaming

### DISCUSSION THREAD

**Discussion by PepeDest** [ant-media/Ant-Media-Server#5776] [Q&A]
Hi, i've been using ant media server on a linode server for a 24/7 stream and everything went fine and smooth, awesome product.
However I moved everything on my personal home server which is running windows server 2022 21H2, i am currently running Ant Media Server on a hypervisor virtual machine which is running Ubuntu 18.04 LTS. 
I've done this:
- Start an Ant Media community server via linux command line
- Wait for server to complete its process by watching the terminal
- Visit site via IP:Port to create and use login for first time
- Go to Applications > LiveApp > New Stream > Playlist
- Name the playlist and click create
- Go to the VoD tab > Upload VoD > Choose File > Grab an mp4 from local system > Click save > Wait until success message > Click OK
Copy VoD ID > Go back to Live Streams Tab > Click Pencil to Edit Live Stream > Click Add Playlist Item > Within the 'Playlist URL input' enter: [protocol]://[ip]:[port]/LiveApp/streams/[VoD ID].mp4 > See Playlist added success toast message at top right
- Visit YouTube, choose to go live > Collect stream URL & stream key
- Back on Ant Media Server, click burger menu to left of created live stream > Edit RTMP Endpoints > Within input box enter: [youtube stream url]/[youtube stream key] > Click 'Add RTMP Endpoint' > Scroll down, click Close
- Click burger menu to right of created live stream again > Click 'Start Broadcast' > See success toast to top right of screen > Wait for Status to show 'Broadcasting'
- Test to see if working on server by clicking play button to the right of the created live stream > it should display a new modal like screen playing your uploaded mp4 video
Go back to YouTube live page, and you should see the stream working, but instead I see the image here
![image](https://github.com/ant-media/Ant-Media-Server/assets/149820226/30fb3701-5e9e-4649-a607-6ed33f81878d)
i tried doing again all the procedure but nothing changed, what else i should try ?

**Comment by tremzo** [comment]
Can you translate what that message says into English and also provide some Ant Media Server logs? Are you seeing your recording play in Ant Media Server just not in Youtube as RTMP endpoint?

**Reply by PepeDest** [reply]
thanks you in advance for the help!
![image](https://github.com/ant-media/Ant-Media-Server/assets/149820226/4e15d348-c770-4ce5-bbee-5d4541e8a5b3)
this the message in english, as soon as i am back at home i will provide logs. I'm seeing the recording trough ant media server and not in the youtube RTMP endpoint.

**Reply by PepeDest** [reply]
```
2023-11-06 21:28:22,649 [vert.x-worker-thread-20] WARN  io.antmedia.muxer.RtmpMuxer - Could not open output url: rtmp://a.rtmp.youtube.com/live2/exq7-xxxx-xxxx-xxxx-xxxx
2023-11-06 21:28:22,649 [vert.x-worker-thread-20] INFO  io.antmedia.muxer.MuxAdaptor - Endpoint status updated to failed  for streamId: sSsFfabt2wem3kNQ1699302154564 for url: rtmp://a.rtmp.youtube.com/live2/exq7-xxxx-xxxx-xxxx-xxxx
2023-11-06 21:28:22,649 [vert.x-worker-thread-20] ERROR io.antmedia.muxer.RtmpMuxer - Cannot initializeOutputFormatContextIO for rtmp endpoint:rtmp://a.rtmp.youtube.com/live2/exq7-xxxx-xxxx-xxxx-xxxx
```
i am not very practical but in the log log seems the most important

**Reply by Mohit-3196** [reply]
Hi Pietro,
How are you?

This is Mohit, we were discussing this earlier. Sorry for not being able to get back to you over the weekend.
Let's continue the discussion here, I was trying to publish a RTMP stream to your server but I couldn't do it. Is there are firewall restrictions on your server?
If yes, can you please check and enable port 1935?

Thank you

**Reply by PepeDest** [reply]
hi! thanks for the answer
we got some firewall but the port 1935 is enabled on the ubuntu hypervisor

**Reply by Mohit-3196** [reply]
Hi @PepeDest,
Thank you.
How about the outbound rules, are they open?
Can you please check if you can publish a RTMP stream to the server?
You can use OBS to push the RTMP stream https://antmedia.io/docs/guides/publish-live-stream/rtmp/publish-with-obs/

Thank you

**Reply by PepeDest** [reply]
i've checked and i can publish a RTMP stream with obs on ant media server

**Reply by Mohit-3196** [reply]
Hi @PepeDest,
Thank you for the confirmation.

Can we schedule a meet and check this issue together?
Let me know a suitable time for you. My timezone is IST.

Thank you.

**Reply by PepeDest** [reply]
here are the outbound rules
![sd](https://github.com/ant-media/Ant-Media-Server/assets/149820226/3eed1792-15fc-44a1-a6e6-abcacbe0f423)

**Reply by Mohit-3196** [reply]
OK,
Let's meet on Wednesday and we'll check.
